pinot-core/src/main/java/com/linkedin/pinot/core/startree/StarTreeDataSorter.java

```diff
@@ -207,15 +206,37 @@ public void remove() {
         byte[] dimBuff = new byte[dimensionSizeInBytes];
         byte[] metBuff = new byte[metricSizeInBytes];
 
-        mappedByteBuffer.copyTo(startOffset + (pointer * totalSizeInBytes), dimBuff, 0, dimensionSizeInBytes);
+        copyTo(startOffset + (pointer * totalSizeInBytes), dimBuff, 0, dimensionSizeInBytes);
```

Comment:
No, I meant that since 'pointer' and 'totalSizeInBytes' are both integer, their multiplication can cause integer overflow unless one of them is upcasted to long. Filed PR: https://github.com/linkedin/pinot/pull/1664 for the same.